app/src/main/java/com/example/moviedetails/SearchableActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.moviedetails
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SearchView
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.moviedetails.network.MovieApi
+import com.example.moviedetails.network.MovieRepository
+import com.example.moviedetails.network.RetrofitClient
+import kotlinx.android.synthetic.main.activity_searchable.*
+
+class SearchableActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_searchable)
+
+        val searchRV = findViewById<RecyclerView>(R.id.searchResults)
+        val viewModel = ViewModelProvider(this)[SearchViewModel::class.java]
+        val movieRepository =
+            MovieRepository(RetrofitClient.getClient().create(MovieApi::class.java))
+        searchRV.layoutManager =
+            LinearLayoutManager(this).apply { orientation = LinearLayoutManager.VERTICAL }
+
+        searchBar.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                viewModel.searchMovie(query, movieRepository)
+                return false
+            }
+
+            override fun onQueryTextChange(newText: String): Boolean {
+                return false
+            }
+        })
+
+        viewModel.listOfSearchedMoviesByTitle.observe(this) {
+            searchRV.adapter = MovieAdapter(it)
```

Comment:
We should not set new adapter for evey data change. instead we should try to refresh the data
Can you check `DiffUtil`, which will enable us to update item which are only changed.